extension/extensiontest/statuswatcher_extension.go

```diff
@@ -21,7 +22,7 @@ func NewStatusWatcherExtensionCreateSettings() extension.Settings {
 
 // NewStatusWatcherExtensionFactory returns a component.ExtensionFactory to construct a status watcher extension.
 func NewStatusWatcherExtensionFactory(
-	onStatusChanged func(source *component.InstanceID, event *component.StatusEvent),
+	onStatusChanged func(source *pipeline.InstanceID, event *component.StatusEvent),
```

Comment:
This is also a breaking change, but if we end up not having a type alias for `component.InstanceID`, users of this API are going to have a build time error regardless so they will be 'notified' that way. We can list it in the API changelog as a breaking change as well.